index.js

```diff
@@ -15,8 +15,29 @@ export default function decompressResponse(response) {
 
 	const checker = new TransformStream({
 		transform(data, _encoding, callback) {
+			if (isEmpty === false) {
+				return callback(null, data);
+			}
+
 			isEmpty = false;
 
+			const decompressStream = contentEncoding === 'br'
+				? zlib.createBrotliDecompress()
+				: ((contentEncoding === 'deflate' && data.length > 0 && (data[0] & 0x08) === 0) // eslint-disable-line no-bitwise
+					? zlib.createInflateRaw()
+					: zlib.createUnzip());
+
+			decompressStream.once('error', error => {
+				if (isEmpty && !response.readable) {
+					finalStream.end();
+					return;
+				}
+
+				finalStream.destroy(error);
+			});
+
+			checker.pipe(decompressStream).pipe(finalStream);
```

Comment:
I don't think the stream piping logic should be inside the `transform` function.

Reply:
We need to look at the first byte before determining the correct compression. So we're doing this inside the transform, when we have the first chunk.

Reply:
At least move it into a separate function that could be placed on line 15. I think that would make it more obvious what's happening.

Reply:
I would also add a short code comment about:

>  look at the first byte before determining the correct compression

Reply:
Refactored out